presto-orc/src/main/java/com/facebook/presto/orc/metadata/statistics/MapStatistics.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.metadata.statistics;
+
+import org.openjdk.jol.info.ClassLayout;
+
+import java.util.List;
+import java.util.Objects;
+
+import static com.facebook.presto.orc.metadata.statistics.StatisticsHasher.Hashable;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class MapStatistics
+        implements Hashable
+{
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(MapStatistics.class).instanceSize();
+    private final List<MapStatisticsEntry> entries;
```

Comment:
Why is this modeled as List<MapStatisticsEntry> ? Should it be Map<> ? The caller has to manage the no duplicate entries here.

Reply:
This is modeled after the orc proto, which models it as list of Key+Value entries, to simplify conversion and to reduce load. 

If we model it as a map, then both reader and writer would have to convert it to map, which is a bad move performance wise, especially given that many tables/files have 11-12k+ keys just in one column. 

If we think about it, then it's modeled the same way as the map writer, which doesn't check key uniqueness and delegates it to the client. 

Unique keys are also an intrinsic property of the MapFlatColumnWrite.